pyproject.toml

```diff
@@ -153,34 +153,21 @@ pythonVersion = "3.8"
 typeCheckingMode = "basic"
 include = ["pandas", "typings"]
 exclude = ["pandas/tests", "pandas/io/clipboard", "pandas/util/version"]
+# enable subset of "strict"
+reportDuplicateImport = true
+reportInvalidStubStatement = true
```

Comment:
enable `reportInvalidStubStatement` to catch default arguments in pyi files.

Unrelated: also simplify pyright config. Initially, I disabled a lot and used the "strict" rule set - but since some time we have been using "basic", so we don't need all of the `... = false` anymore. Also enable a few rules that were active when we used "strict".

Reply:
nice. These changes alone did not reveal the unused imports that are also removed in this PR. You found these manually? Which options are needed to catch these automatically? Pylance reports for instance `"NaTType" is not accessed`, but running `pre-commit run -a --hook-stage manual pyright` with this new config only produced the 2 `error: Default values in stub files...` errors.

Reply:
I was running flake8 on all pyi files and then ignoring many errors that are not related to pyi files. The better option is to install flake8-pyi (I tried that but I'm not unfamiliar enough with pre-commit).

Reply:
> The better option is to install flake8-pyi

sgtm (as followup) or flake8 with a config that ignores E302 and E704 (and probably others)

Reply:
> Pylance reports for instance `"NaTType" is not accessed`

I assume they enable reportUnusedImport/Class/function/Variable. If I enable them we seem to get many false positives (might be a bug in pyright or pyright determines that the only place where something is used cannot be reached, potentially because of incorrect type annotations).